Doc/tutorial/datastructures.rst

```diff
@@ -675,8 +675,8 @@ to a variable.  For example, ::
    >>> non_null
    'Trondheim'
 
-Note that in Python, unlike C, assignment cannot occur inside expressions. C
-programmers may grumble about this, but it avoids a common class of problems
+Note that in Python, unlike C, assignment inside expressions must be done
+explicitly with the ``:=`` operator. This avoids a common class of problems
```

Comment:
Minor change to mention that it's known as the "walrus operator":
```suggestion
explicitly with the walrus operator ``:=`` . This avoids a common class of problems
```
This format was also used in https://docs.python.org/3/faq/design.html#why-can-t-i-use-an-assignment-in-an-expression.